src/card/authentication.js

```diff
@@ -0,0 +1,37 @@
+const { Validate, Auth, fetchWithHTTPErr } = require('../utils');
+const errors = require('../errors');
+
+function createAuthentication(data) {
+  return new Promise((resolve, reject) => {
+    try {
+      const compulsoryFields = ['amount', 'tokenID'];
+      const missingFields = Validate.requiredFields(compulsoryFields).validate(
+        data,
+      );
+      if (missingFields.length > 0) {
+        let message = Validate.missingFieldsErrMsg(missingFields);
+        reject({ status: 400, code: errors.API_VALIDATION_ERROR, message });
+      }
+
+      fetchWithHTTPErr(
+        `${this.API_ENDPOINT}/credit_card_tokens/${data.tokenID}/authentications`, // eslint-disable-line max-len
+        {
+          method: 'POST',
+          headers: {
+            Authorization: Auth.basicAuthHeader(this.opts.publicKey),
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify({ amount: data.amount }),
+        },
+      )
+        .then(resolve)
+        .catch(reject);
+    } catch (e) {
+      reject({ status: 418, code: errors.JS_ERROR, message: e.message });
```

Comment:
lol what's up with the teapot code :laughing:

Reply:
HAHAH This part is not a HTTP error so I was just being cheeky to make a consistent error response